core/monitor.py

```diff
@@ -176,6 +176,8 @@ def run(self):
         # to this point. It's the job of the subclass to decide where
         # to go from here.
         timestamp_obj = self.timestamp()
+        # commit it now in case something fails during the run.
+        self._db.commit()
```

Comment:
Is this commit  required if we are forcing the timestamp object in the `except` block?

Reply:
If there is a failure expected later on, do we want partial data saved?

Reply:
That's a good point.  Good catch @RishiDiwanTT .  I'll remove the commit.